sources/packages/backend/apps/workers/src/controllers/supporting-user/supporting-user.controller.ts

```diff
@@ -61,7 +69,33 @@ export class SupportingUserController {
         (supportingUser) => supportingUser.id,
       );
       jobLogger.log("Created supporting users.");
-      return job.complete({ createdSupportingUsersIds });
+      return this.dataSource.transaction(
+        async (transactionalEntityManager: EntityManager) => {
+          const application = await this.applicationService.getApplicationById(
+            job.variables.applicationId,
+          );
+          const supportingUserType =
+            job.variables.supportingUsersTypes[0] === SupportingUserType.Partner
+              ? "partner"
+              : job.variables.supportingUsersTypes.length > 1
+              ? "parents"
+              : "parent";
```

Comment:
My apologies on suggesting the double ternary operator. The sonar is complaining about it. 

Can you please move this logic using if into private method ?

```ts
private getNotificationSupportingUserType(
    supportingUsers: SupportingUserType[],
  ): NotificationSupportingUserType {
    const [supportingUser1, supportingUser2] = supportingUsers;
    // If the supporting user is partner, only one supporting user can be present.
    if (supportingUser1 === SupportingUserType.Partner) {
      return "partner";
    }
    return supportingUser2 ? "parents" : "parent";
  }
}
```